extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php

```diff
@@ -443,6 +446,7 @@ public function execute() {
 			}
 
 			header( "X-Redirected-By-WF: NotPrimary" );
+			header( "Vary: Cookie");
```

Comment:
One last question here is how long Fastly is caching this? I'm mostly thinking if we should make sure the cache is truly private for logged-in users so we don't cache a lot of versions for each access_token. If it's a very short cache, it's probably not a big concern at this point given the redirect URLs are probably relatively rare to be clicked by many logged-in users.

Reply:
ping: @ljagiello could you check this one?

Reply:
After offsite talk we decided to vary on source protocol (bda16cdff15) instead of `Cookie` (which is a bad idea). It should cover most of our cases.

Reply:
We already vary on source protocol on the Fastly level: https://wikia-inc.atlassian.net/browse/OPS-11119 so this should be unnecessary in theory.

The reason to vary on cookie is because the behaviour is different depending on if you're logged in or anonymous. Anons should go HTTPS -> HTTP, and logged in users should go HTTPS -> HTTPS (but they currently have two redirects for aliases: HTTPS -> HTTP -> HTTPS). If this redirect is cached and we don't vary on cookie/logged-in, we may move the double redirect to anons in some cases, i.e. if they request the alias over HTTPS and the redirect is currently cached, they will be redirected HTTPS -> HTTPS -> HTTP. It also means [this check](https://github.com/Wikia/app/blob/bda16cdff1580/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L430) wouldn't work very well.

Providing we make sure the logged-in request/redirect is not cached, there should only be two variations of the request for Fastly to handle (anons who have the cookie header stripped because they don't have any of our auth-related cookies set, and logged-in users for whom the cache should be private (so no versions should be stored in Fastly)).

If this is not possible, and the redirect is being cached on Fastly, I think we should consider abandoning this, as it's better to have a slightly broken redirect experience for logged-in users (and get them to update their imports), than potentially start giving double redirects to any HTTPS URLs anons (and in this case Google and other crawlers) may encounter, even if it's relatively unlikely right now (though it will become more likely as logged-in users opt-in and start pasting HTTPS URLs in various places).

What do you think?